### PR TITLE
[#261] [mezzio-toggle] Make as optional the installation of some pheature packages

### LINKED ISSUE #261: [mezzio-toggle] Make as optional the installation of some pheature packages
Epic: #24 Mezzio Toggle
Package: https://github.com/pheature-flags/mezzio-toggle

Description:

Right now, if you want to run Pheature Flags using the Mezzio Toggle, you need to install all packages of the Pheature framework. We should make them optional, adding some checks to the `ToggleConfigProvider` to add services only if they've been previously installed.

### PR DESCRIPTION
**Pull request by xserrat**
It closes #261

### CODE REVIEW

**Comment by xserrat** [comment]
- [x] Missing ConfigProvider tests
- [x] Remove extra packages from mezzio-toggle `composer.json`

File: packages/toggle-model/test/Fixtures/Configuration.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+use Pheature\Model\Toggle\SegmentFactory;
+use Pheature\Model\Toggle\StrategyFactory;
+
+final class Configuration
+{
+    public static function create(): array
+    {
+        return [
+            'dependencies' => [
```
Comment (kpicaza) [inline packages/toggle-model/test/Fixtures/Configuration.php]: Maybe these test configuration classes can be avoided by adding the config array as constants in the test classes? 
Or we are using each test configuration class in more than one test?

**Review by kpicaza — COMMENTED** [review]
:clap::clap::clap: Good job ;-D I left a question related to Test Configuration classes

File: packages/toggle-model/test/Fixtures/Configuration.php (hunk 1 above)
Comment (xserrat) [reply packages/toggle-model/test/Fixtures/Configuration.php]: done!

**Review by kpicaza — APPROVED** [review]
:ship::ship::ship:

**Pull request merged**